src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            reSize();
+        }
+        int keyHash = (key == null) ? 0 : key.hashCode();
+        Node<K, V> newNode = new Node<>(keyHash, key, value, null);
+        int index = hash(key);
```

Comment:
The `hash` method should ensure that it handles negative hash codes correctly. Consider using `Math.abs(key.hashCode()) % table.length` to avoid negative indices.